test/parallel/test-cluster-master-kill.js

```diff
@@ -21,10 +21,17 @@ if (cluster.isWorker) {
   // terminate the cluster process
   worker.once('listening', function() {
     setTimeout(function() {
-      process.exit(0);
+/* Cluster.disconnect() will exercise a different 'graceful' shutdown path.
+   From the perspective of the worker, closing the channel is equivalent
+   to the parent calling process.exit(0). */
+      worker.process._channel.close();
```

Comment:
I don't object, but this does use undocumented APIs, and the last test didn't. That can be annoying when trying to rearrange things that are supposed to be internal.

Why not just change the check at the end to not do a single check after some randomly chosen time interval, but to run the check every half-second, indefinitely, until it passes or the runner kills this?